R/simpleCache.R

```diff
@@ -12,254 +13,268 @@
+			parse = TRUE
+			warning(strwrap("Detected a character instruction; consider wrapping
+			in {} instead of quotes."))
 		} else {
-			parse=FALSE;
+			parse = FALSE
 		}
 	}
 	if(!is.null(cacheSubDir)) {
-		cacheDir=paste0(cacheDir, cacheSubDir);
+		cacheDir = paste0(cacheDir, cacheSubDir)
 	}
 	if (is.null(cacheDir)) {
-		message("You must set global option RCACHE.DIR with setSharedCacheDir(), or specify a cacheDir parameter directly to simpleCache().");
-		return(NA);
+		message(strwrap("You must set global option RCACHE.DIR with setSharedCacheDir(),
+		or specify a cacheDir parameter directly to simpleCache()."))
+		return(NA)
 	}
-	if(! "character" %in% class(cacheName)) {
-		stop("simpleCache expects the cacheName variable to be a character vector.");
+	if(!"character" %in% class(cacheName)) {
+		stop("simpleCache expects the cacheName variable to be a character
+		vector.")
 	}
-  
 	
-	cacheDir=enforceTrailingSlash(cacheDir);
+	cacheDir = enforceTrailingSlash(cacheDir)
 	if (!file.exists(cacheDir)) {
-		dir.create(cacheDir, recursive=TRUE);
+		dir.create(cacheDir, recursive=TRUE)
 	}
 	cacheFile = paste0(cacheDir, cacheName, ".RData")
 	lockFile = paste0(cacheDir, cacheName, ".lock")
-	if (ignoreLock) { #otherwise use a trycatch... ?
-		on.exit(file.remove(lockFile));
+	if (ignoreLock) {
+		# remove the lock file when this function call is complete.
+		on.exit(file.remove(lockFile))
 	}
-	submitted=FALSE;
-	#check if cache exists in any provided search environment.
-	searchEnvir = append(searchEnvir, ".GlobalEnv");	#assume global env.
-	cacheExists = FALSE; cacheWhere = NULL;
+	submitted = FALSE
+	# Check if cache exists in any provided search environment.
+	searchEnvir = append(searchEnvir, ".GlobalEnv")  # Assume global env.
+	cacheExists = FALSE
+	cacheWhere = NULL
 	for ( curEnv in searchEnvir ) {
 		if(exists(cacheName, where=get(curEnv))) {
-			cacheExists = TRUE;
-			cacheWhere = curEnv;
-			break;
+			cacheExists = TRUE
+			cacheWhere = curEnv
+			break
 		}
 	} #for
 	
-	ret = NULL; # The default; in case the cache construction fails.
+	ret = NULL # The default, in case the cache construction fails.
 
 	if(cacheExists & !reload & !recreate) {
-		message("::Object exists (in ", cacheWhere, ")::\t", cacheName);
-		#return(get(cacheName));
-		#return();
-		ret = get(cacheName, pos=get(cacheWhere));
+		message("::Object exists (in ", cacheWhere, ")::\t", cacheName)
+		#return(get(cacheName))
+		#return()
+		ret = get(cacheName, pos=get(cacheWhere))
 	} else if (file.exists(lockFile) & !ignoreLock) {
 		message("::Cache processing (lock file exists)::\t", lockFile)
 		#check for slurm log...
-		if (!is.null(slurmParams)) { slurmLog = paste0(slurmParams$hpcFolder, "/", cacheName, ".log"); message(slurmLog); utils::tail(readLines(slurmLog), 10) }
 
-		return ();
+		if (!is.null(batchRegistry)) {
+			# Grabbing log from batchtools
+			# 1 is the job id.
+			message(paste(batchtools::getLog(1, reg=registry), collapse="\n"))
```

Comment:
should this `reg=registry` or `reg=batchRegistry`? see see [`check()`](https://github.com/databio/simpleCache/pull/5#issuecomment-318659752) results